Installation not working with cmake version 2.8.11

Hi there,

Using cmake version 2.8.11, `cmake -G "Unix Makefiles"` resulted in:

CMake Error at CMakeLists.txt:128 (set):
  Syntax error in cmake code at

    /home/eindik/flatbuffers/CMakeLists.txt:129

  when parsing string

    ${CMAKE_CXX_FLAGS} -Wunused-result -Werror=unused-result \
                            -Wunused-parameter -Werror=unused-parameter

  syntax error, unexpected cal_SYMBOL, expecting $end (128)


To fix, I removed the line break between lines 129 and 130 of CMakeLists.txt.
Apparently, cmake 2.8 doesn't support that sort of line breaking.

We may want to update `cmake_minimum_required(VERSION 2.8)` to something more recent.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -126,8 +126,7 @@ elseif(CMAKE_COMPILER_IS_GNUCXX)
         "${CMAKE_CXX_FLAGS} -faligned-new")
     endif()
     set(CMAKE_CXX_FLAGS
-      "${CMAKE_CXX_FLAGS} -Wunused-result -Werror=unused-result \
-                          -Wunused-parameter -Werror=unused-parameter")
+      "${CMAKE_CXX_FLAGS} -Wunused-result -Werror=unused-result -Wunused-parameter -Werror=unused-parameter")
   endif()
 
   # Certain platforms such as ARM do not use signed chars by default

